### ISSUE TITLE
u32 variable name conflicts with u32 library type

### ISSUE THREAD

**Issue by haata** [KDAB/cxx-qt#677]
I'm a bit at a loss on how to resolve this one.

```bash
  cargo:warning=/home/hyatt/Source/hid-io/OpenRGBHidioPlugin/target/debug/build/openrgb-sample-plugin-568e03b7a7ba2cc4/out/cxx-qt-lib/std_types.h:25:18: error: ‘const int u32’ redeclared as different kind of entity

  cargo:warning=   25 | static const int u32 = qRegisterMetaType<::std::uint32_t>("::std::uint32_t");

  cargo:warning=      |                  ^~~

  cargo:warning=OpenRGB/i2c_smbus/i2c_smbus.h:22:25: note: previous declaration ‘typedef unsigned int u32’

  cargo:warning=   22 | typedef unsigned int    u32;
  ```
  
I'm trying to wrap an existing plugin library which uses u32 as a type in an i2c library (quite extensively too).
I believe this source code is pulled from C (looks familiar, probably Linux kernel) https://gitlab.com/CalcProgrammer1/OpenRGB/-/blob/master/i2c_smbus/i2c_smbus.h?ref_type=heads.

Are there any options for me on the cxx-qt side without having to suggest major patches to OpenRGB?

**Comment by haata** [comment]
Haven't gotten through all my errors yet, but I can make progress by fixing the library to use defines (and then undef them after the initial include).

```diff
diff --git a/i2c_smbus/i2c_smbus.h b/i2c_smbus/i2c_smbus.h
index 6f87eb61..a9b58cf9 100644
--- a/i2c_smbus/i2c_smbus.h
+++ b/i2c_smbus/i2c_smbus.h
@@ -16,10 +16,10 @@
 #include <condition_variable>
 #include <mutex>

-typedef unsigned char   u8;
-typedef unsigned short  u16;
-typedef unsigned int    u32;
-typedef int             s32;
+#define u8 unsigned char
+#define u16 unsigned short
+#define u32 unsigned int
+#define s32 int

 #ifdef _WIN32
 ```
 
 Hopefully this will be a palatable upstream patch.

**Comment by ahayzen-kdab** [comment]
Thanks for this report !

I wonder if this has been fixed in `main` with what will become 0.6. As we change changed the std_types from being in the global namespace ...

https://github.com/KDAB/cxx-qt/blob/0.5.x/crates/cxx-qt-lib-headers/include/std_types.h

To now being in the `rust::cxxqtlib1` namespace

https://github.com/KDAB/cxx-qt/blob/main/crates/cxx-qt-build/src/std_types_qt5.cpp

I'll check this next week to ensure it does solve the issue :-)

**Comment by ahayzen-kdab** [comment]
So looks like #534 solved this with https://github.com/KDAB/cxx-qt/pull/534/commits/94aef9a3e48ec27ffc0e855ea4daa4a539883bcb therefore i'm closing this issue. This will end up in the 0.6 release (which should hopefully be soon).

Please feel free to open new issues for problems you find :-)
